EIPS/eip-7496.md

```diff
@@ -0,0 +1,216 @@
+---
+eip: 7496
+title: NFT Dynamic Traits
+description: Extension to ERC-721 and ERC-1155 for dynamic onchain traits
+author: Adam Montgomery (@montasaurus), Ryan Ghods (@ryanio), 0age (@0age), James Wenzel (@jameswenzel), Stephan Min (@stephankmin)
+discussions-to: https://ethereum-magicians.org/t/erc-7496-nft-dynamic-traits/15484
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-28
+requires: 165, 721, 1155
+---
+
+## Abstract
+
+This specification introduces a new interface that extends [ERC-721](./eip-721.md) and [ERC-1155](./eip-1155.md) that defines methods for setting and getting dynamic onchain traits associated with non-fungible tokens. These dynamic traits can be used to represent properties, characteristics, redeemable entitlements, or other attributes that can change over time. By defining these traits onchain, they can be used and modified by other onchain contracts.
+
+## Motivation
+
+Trait values for non-fungible tokens are often stored offchain. This makes it difficult to query and mutate these values in contract code. Specifying the ability to set and get traits onchain allows for new use cases like transacting based on a token's traits or redeeming onchain entitlements.
+
```

Comment:
I'd like to see the motivation expanded a bit. Particularly I'd like to know why we need a general interface for traits instead of using regular `getFoo() returns uint` and `setFoo(uint foo)` style functions.

What is the motivating use case behind this proposal? What type of dapp would need to interact with different tokens and completely arbitrary traits?

Reply:
thanks, added reasoning for the motivating use case and why this over get*/set* style functions